src/server/controllers/user.js

```diff
@@ -109,27 +125,60 @@ const editUser = async (req, res) => {
     res.json({ data: updatedUser });
 };
 
-const createProfile = async (req, res) => {
-    const { userId, profilePicture, location } = req.body;
+const getUserById = async (req, res) => {
+    const id = parseInt(req.params.id, 10);
 
-    const createdProfile = await prisma.profile.create({
-        data: {
-            userId: userId,
-            profilePicture: profilePicture,
-            location: location,
+    let selectedUser = await prisma.user.findUnique({
+        where: { 
+            id: id,
+        }, 
+        include: {
+            profile: true,
         },
-    });
+    })
+
+    if (!selectedUser) {
+        return res.status(SERVER_ERROR.NOT_FOUND.CODE).json({ error: SERVER_ERROR.NOT_FOUND.MESSAGE });
+    }
+
+    selectedUser = removeKeys(selectedUser, KEYS.PASSWORD);
+
+    res.status(SERVER_SUCCESS.OK.CODE).json({ data: selectedUser });
+} 
+
+const createProfile = async (req, res) => {
+    const userId = parseInt(req.body.userId, 10);
```

Comment:
Consider using Number () rather than parseInt() which has strange behaviour if you're not careful